web/packages/teleterm/electron.vite.config.mts

```diff
@@ -0,0 +1,169 @@
+    projects: [resolve(rootDirectory, 'tsconfig.json')],
+  });
+
+  const commonPlugins = [
+    externalizeDepsPlugin({ exclude: externalizeDeps }),
+    tsConfigPathsPlugin,
+  ];
+
+  const config: UserConfig = {
+    main: {
+      build: {
+        outDir: resolve(outputDirectory, 'main'),
+        rollupOptions: {
+          input: {
+            index: resolve(__dirname, 'src/main.ts'),
+            sharedProcess: resolve(
+              __dirname,
+              'src/sharedProcess/sharedProcess.ts'
+            ),
+            agentCleanupDaemon: resolve(
+              __dirname,
+              'src/agentCleanupDaemon/agentCleanupDaemon.js'
+            ),
+          },
+          onwarn,
+          output: {
+            manualChunks,
+          },
+        },
+      },
+      plugins: commonPlugins,
+    },
+    preload: {
+      build: {
+        outDir: resolve(outputDirectory, 'preload'),
+        rollupOptions: {
+          input: {
+            index: resolve(__dirname, 'src/preload.ts'),
+          },
+          onwarn,
+          output: {
+            manualChunks,
+          },
+        },
+      },
+      plugins: commonPlugins,
+    },
+    renderer: {
+      root: '.',
+      build: {
+        outDir: resolve(outputDirectory, 'renderer'),
+        rollupOptions: {
+          input: {
+            index: resolve(__dirname, 'index.html'),
+          },
+        },
+      },
+      server: {
+        host: '0.0.0.0',
+        port: 8080,
+        fs: {
+          allow: [rootDirectory, '.'],
+        },
+      },
+      plugins: [
+        react({
+          plugins: [
+            [
+              '@swc/plugin-styled-components',
+              getStyledComponentsConfig(env.mode),
+            ],
+          ],
+        }),
+        cspPlugin(getConnectCsp(env.mode === 'development')),
+        tsConfigPathsPlugin,
+      ],
+      define: {
+        'process.env': { NODE_ENV: process.env.NODE_ENV },
+      },
+    },
+  };
+
+  if (env.mode === 'development') {
+    if (process.env.VITE_HTTPS_KEY && process.env.VITE_HTTPS_CERT) {
+      config.renderer.server.https = {
+        key: readFileSync(process.env.VITE_HTTPS_KEY),
+        cert: readFileSync(process.env.VITE_HTTPS_CERT),
+      };
+    } else {
+      const certsDirectory = resolve(rootDirectory, 'web/certs');
+
+      if (!existsSync(certsDirectory)) {
+        throw new Error(
+          'Could not find SSL certificates. Please follow web/README.md to generate certificates.'
+        );
+      }
+
+      const keyPath = resolve(certsDirectory, 'server.key');
+      const certPath = resolve(certsDirectory, 'server.crt');
+
+      config.renderer.server.https = {
+        key: readFileSync(keyPath),
+        cert: readFileSync(certPath),
+      };
+    }
+  }
+
+  return config;
+});
+
+export { config as default };
+
+function onwarn(warning, warn) {
+  // ignore Vite complaining about protobufs using eval
+  if (warning.code === 'EVAL') {
```

Comment:
I don't see any warning related to `eval` if I remove this. I guess we don't need it.

Reply:
Ah, must've been from the other TypeScript protobuf gen library I had tried. I'll remove!